src/core/public/chrome/nav_controls/nav_controls_service.ts

```diff
@@ -104,7 +123,19 @@ export class NavControlsService {
         navControlsExpandedCenter$.next(
           new Set([...navControlsExpandedCenter$.value.values(), navControl])
         ),
-
+      registerRightNavigation: (props: RightNavigationProps) => {
+        const nav = {
+          order: props.order,
+          mount: mountReactNode(
+            React.createElement(RightNavigationButton, {
+              ...props,
+              http,
+              application,
+            })
+          ),
+        };
+        return navControlsRight$.next(new Set([...navControlsRight$.value.values(), nav]));
```

Comment:
Nit: why don't we just call the `registerRight` function? The registerRight logic may change in the future.

Reply:
Updated.